cicd/forgeops-tests/forgeops-tests.py

```diff
@@ -16,7 +21,42 @@
 # Insert config folder (rank 1) to python path
 sys.path.insert(0, os.path.join(root_dir, 'config'))
 
-import pytest
+
+def set_allure_environment_props(filename):
+    # Get os environment properties as dictionary
+    environment_properities = dict(os.environ)
+
+    # Read in the properties currently specified in environment.properties
+    current_properties = {}
+    if os.path.exists(filename):
+        with open(filename, 'r') as file:
+            for line in file:
+                line = line.rstrip()
+                if "=" not in line: continue
+                if line.startswith("#"): continue
+                key, value = line.split("=", 1)
+                current_properties[key] = value
+
+    # Remove properties contained previously in current properties that are not specified in environment properties
+    for key in list(current_properties.keys()):
+        if key not in environment_properities:
+            del current_properties[key]
+
+    # Iterate through environment properties and if contained in current properties
+    # change current properties value if they differ.  Otherwise, if environment properties
+    # contains a new property beginning with TESTS_ that isn't in current properties, add it.
+    for key, value in environment_properities.items():
+        if key in current_properties:
+            if value != current_properties[key]:
+                current_properties[key] = value
+        if key not in current_properties and key.startswith("TESTS_"):
+            current_properties[key] = environment_properities[key]
+
+    # Update environment.properties with properties for this test run
+    with open(filename, 'w') as file:
+        for key, value in current_properties.items():
+            file.write('%s=%s\n' % (key, value))
```

Comment:
Could this not be simplified to writing the environment variables prefixed with "TESTS" directly to this file?  It seems as though we are only interested in the environment variables.

Reply:
Good point.  I probably over thought this.